homework/index.js

```diff
@@ -30,14 +32,145 @@
+    const updatedOn = new Date(selectedRepository.updated_at).toLocaleString();
+    const repoURL = selectedRepository.html_url;
+
+    const infoContainerDiv = document.getElementById('repo-info-container');
+    const repoInfodiv = createAndAppend('div', infoContainerDiv, {
+      class: 'repo-info',
+    });
+    const div = createAndAppend('div', repoInfodiv);
+    const table = createAndAppend('div', div, { class: 'grid' });
+
+    const tr1 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr1, { text: 'Repository:', class: 'col-50' });
+    const repoNameCol = createAndAppend('div', tr1, { class: 'col-50' });
+    createAndAppend('a', repoNameCol, {
+      href: repoURL,
+      target: '_blank',
+      text: repositoryName,
+    });
+
+    const tr2 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr2, { text: 'Description:', class: 'col-50' });
+    createAndAppend('div', tr2, { text: description, class: 'col-50' });
+
+    const tr3 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr3, { text: 'Forks:', class: 'col-50' });
+    createAndAppend('div', tr3, { text: forks, class: 'col-50' });
+
+    const tr4 = createAndAppend('div', table, { class: 'row' });
+    createAndAppend('div', tr4, { text: 'Updated:', class: 'col-50' });
+    createAndAppend('div', tr4, { text: updatedOn, class: 'col-50' });
+  }
+
+  function repositoryChanged(event) {
+    const infoElement = document.getElementById('repo-info-container');
+    infoElement.innerHTML = '';
+    const repositoryIndex = parseInt(event.target.selectedOptions[0].value, 10);
+    const selectedRepo = repositoryList[repositoryIndex];
+    bindRepositoryDetails(selectedRepo);
+    bindContributions(selectedRepo.contributors_url);
+  }
+
+  function createDropDown(parentElement, repositories) {
+    const topNavDiv = createAndAppend('div', parentElement, { class: 'topnav' });
+    const div = createAndAppend('div', topNavDiv, { class: 'header-container' });
+    const div1 = createAndAppend('div', div, { class: 'col-50' });
+    createAndAppend('a', div1, { text: 'HYF Repositories', href: '#' });
+    const repoDiv = createAndAppend('div', div, { class: 'col-50' });
+    const repoDropDown = createAndAppend('select', repoDiv, {
+      id: 'ddlRepository',
+      class: 'select-css',
+    });
+    createAndAppend('option', repoDropDown, { text: 'Select Repository' });
+    repositories.sort((a, b) => {
+      const x = a.name.toLowerCase();
+      const y = b.name.toLowerCase();
+      if (x < y) {
+        return -1;
+      }
+      if (x > y) {
+        return 1;
+      }
+      return 0;
+    });
+
+    for (let index = 0; index < repositories.length; index++) {
+      const element = repositories[index];
+      createAndAppend('option', repoDropDown, {
+        text: element.name,
+        value: index,
+      });
+    }
+    repoDropDown.onchange = event => repositoryChanged(event);
```

Comment:
You can pass the selected repo directly:

```js
repoDropDown.onchange = () => repositoryChanged(repositories[repoDropDown.value]);
```

Reply:
done